### PR TITLE
[SPARK-49936][BUILD] Upgrade `datasketches-java` to 6.1.1

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims to upgrade `datasketches-java` from 6.0.0 to 6.1.1.

### Why are the changes needed?
The new version is now dependent on `datasketches-memory` 3.x. The full release notes as follows:
- https://github.com/apache/datasketches-java/releases/tag/6.1.0
- https://github.com/apache/datasketches-java/releases/tag/6.1.1
- https://github.com/apache/datasketches-memory/releases/tag/3.0.0
- https://github.com/apache/datasketches-memory/releases/tag/3.0.1
- https://github.com/apache/datasketches-memory/releases/tag/3.0.2

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by LuciferYang** [comment]
wait 6.1.1 release https://github.com/apache/datasketches-java/releases

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ on:
       java:
         required: false
         type: string
-        default: 17
+        default: 21
```
Comment (LuciferYang) [inline .github/workflows/build_and_test.yml:28]: will revert after test

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @LuciferYang .
Merged to master for Apache Spark 4.0.0 on February 2025.

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun

**Pull request closed without merging**